internal/test/e2e/version_compatibility_test.go

```diff
@@ -273,6 +273,152 @@ func TestVersionCompatibility_PluginFieldUpdates(t *testing.T) {
 	}
 }
 
+func TestVersionCompatibility_EnsureTargetFieldsAreNotOverridden(t *testing.T) {
+	cleanup := run.Koko(t)
+	defer cleanup()
+
+	dpCleanup := run.KongDP(kong.GetKongConfForShared())
+	defer dpCleanup()
+	require.NoError(t, util.WaitForKong(t))
+	require.NoError(t, util.WaitForKongAdminAPI(t))
```

Comment:
We should ensure this tests only runs for 3.0.0+.

```suggestion
	require.NoError(t, util.WaitForKongAdminAPI(t))
	
	kongClient.RunWhenKong(t, ">=3.0.0")
```